src/rewriters/DivModRewriter.h

```diff
@@ -36,6 +36,7 @@ class DivModConfig : public DefaultRewriterConfig {
             PTRef modVar = divMod.mod;
             PTRef rewritten = logic.isIntDiv(symRef) ? divVar : modVar;
             if (not inCache) {
+                if (logic.isNonlin(term)) { return term; }
```

Comment:
This is weird. We should not keep terms like `x div y` or `x mod y` in the formula.

Reply:
This terms are being detected by `LASolver` later anyway, but we might use `DivModRewriter` before the nonlin solver potentially, so why not?

Reply:
Throwing exception for now, added issue to handle it propperly #815